src/GraphQL/Execution/Nodes/ObjectExecutionNode.cs

```diff
@@ -33,7 +33,7 @@ public IObjectGraphType GetObjectGraphType(ISchema schema)
         {
             var objectGraphType = GraphType as IObjectGraphType;
 
-            if (GraphType is IAbstractGraphType abstractGraphType && IsResultSet)
+            if (GraphType is IAbstractGraphType abstractGraphType)
```

Comment:
I'm not exactly sure when this executes (unions or interfaces???), but I ran all the tests and the result was always set.